src/lib/parsers/Args.ts

```diff
@@ -783,12 +58,42 @@ export class Args {
 	}
 }
 
-export interface ArgsJson {
-	message: Message<boolean>;
-	command: MessageCommand;
-	commandContext: MessageCommand.RunContext;
+export interface ArgsOptions<T = unknown, K extends keyof ArgType = keyof ArgType>
+	extends Omit<Argument.Context, 'messageOrInteraction' | 'command'> {
+	// Up to the person implementing if this should always be required or only required for chat commands, but this should
+	// always be required for chat commands, and be used to find the starting point for our parsing
+	name: string;
+	type: IArgument<T> | K;
+	minimum?: number;
+	maximum?: number;
+	inclusive?: boolean;
 }
 
+export interface RepeatArgsOptions extends ArgsOptions {
+	/**
+	 * The maximum amount of times the argument can be repeated.
+	 * @default Infinity
+	 */
+	times?: number;
+}
+
+export interface PeekArgsOptions<T = unknown, K extends keyof ArgType = keyof ArgType> extends Omit<ArgsOptions, 'type'> {
+	name: string;
+	type: () => Awaitable<Argument.Result<T>> | K;
```

Comment:
Any reason for having this interface? 👀 Also, if this is intended, you should add it to the peekResult/peek type in the Args class

Reply:
This interface is used for peek/peekResult, I've added it to Args.